### PR TITLE
Removes CacheConfiguration in favor of setting directories directly on BuildConfiguration.

### PR DESCRIPTION
**Pull request by coollog**
This helps with implementing `JibContainerBuilder#containerize`. `CacheConfiguration` seems like an unnecessary and clumsy extra class to use to just configure basic cache directory paths.

### CODE REVIEW

File: jib-core/src/main/java/com/google/cloud/tools/jib/api/Containerizer.java [hunk 1]
```diff
@@ -28,6 +32,14 @@
 // TODO: Add tests once JibContainerBuilder#containerize() is added.
 public class Containerizer {
 
+  /**
+   * The default directory for caching the base image layers, in {@code [user cache
+   * home]/google-cloud-tools-java/jib}.
+   */
+  // TODO: Reduce scope once plugins are migrated to use the new Jib Core API.
```
Comment (briandealwis) [inline jib-core/src/main/java/com/google/cloud/tools/jib/api/Containerizer.java]: Seems an odd comment when this is a newly-introduced field?

Comment (coollog) [reply jib-core/src/main/java/com/google/cloud/tools/jib/api/Containerizer.java]: This is currently used to set the default in `PluginsConfigurationProcessor`. However, once the plugins are migrated to use the new API (and this class), it won't need to use this field to set defaults as this class sets the default itself.

File: jib-core/src/main/java/com/google/cloud/tools/jib/api/Containerizer.java [hunk 2]
```diff
@@ -106,8 +139,23 @@ TargetImage getTargetImage() {
     return Optional.ofNullable(executorService);
   }
 
-  Optional<CacheConfiguration> getCacheConfiguration() {
-    return Optional.ofNullable(cacheConfiguration);
+  Path getBaseImageLayersCacheDirectory() {
+    return baseImageLayersCacheDirectory;
+  }
+
+  Path getApplicationLayersCacheDirectory() throws CacheDirectoryCreationException {
```
Comment (chanseokoh) [inline jib-core/src/main/java/com/google/cloud/tools/jib/api/Containerizer.java]: Who's calling this? Can we guarantee this is not called concurrently?

Comment (coollog) [reply jib-core/src/main/java/com/google/cloud/tools/jib/api/Containerizer.java]: This is to-be-called by `JibContainerBuilder#containerize`. None of the API classes are thread-safe and should not be called concurrently.

**Pull request merged**